doc/developers-guide/Stanza-routing.md

```diff
@@ -1,76 +1,74 @@
 # Route of a message through the system
 
-Let's examine the flow of a message sent from user A to user B, both of whom are served by the same domain.
+Let's examine the flow of a message sent from Alice to Bob, both of whom are served by the same domain and connected to the server.
 
-Note that hooks are called at various stages of routing - they perform many tasks, in fact, many MongooseIM functionalities are implemented through hooks & handlers. 
+Note that hooks are called at various stages of routing - they perform many tasks, and many MongooseIM functionalities are implemented through hooks & handlers.
 For a general introduction to hooks, see [Hooks and Handlers](Hooks-and-handlers.md); to get a closer look at a core few, see the [hooks description](hooks_description.md).
 
-## 1. Receiving the stanza
+## 1. Sender's C2S process receives the message
 
-User A's `ejabberd_receiver` receives the stanza and passes it to `ejabberd_c2s`.
+Alice's C2S (client-to-server) process, which is a state machine implemented in the `mongoose_c2s` module, receives data from the TCP socket, and parses each incoming XML element with `exml` to an internal representation of the stanza, which is then processed by the C2S as a subsequent event.
 
-## 2. Call to `user_send_packet`
+## 2. Call to `user_send_*` hooks
 
-Upon some minimal validation of the stanza, a hook `user_send_packet` is called.
-This is handled by a couple of modules which subscribe to this hook. 
-Those modules do various complementary tasks, like storing the message in an archive, sending carbon copies, etc.
+Upon some minimal validation of the stanza, the hook `user_send_packet` is called.
+Next, depending on the type of the stanza, one of the following hooks is called:
 
-## 3. Privacy lists and `ejabberd_router:route/3`
+* `user_send_message` for messages,
+* `user_send_presence` for presences,
+* `user_send_iq` for IQ (info/query) stanzas,
+* `user_send_xmlel` for other XML elements.
 
-The stanza is checked against any privacy lists in use and, in the case of being allowed, it will be routed by `ejabberd_router:route/3`. 
-This also takes into account "blocking commands", which are part of the privacy system.
+Each hook can be handled by multiple modules subscribed to it. Those modules do various complementary tasks, like storing the message in an archive, sending carbon copies, checking the stanza against privacy lists etc. See [hooks description](../hooks_description/#hooks-called-for-session_established) for more information.
```

Comment:
I'd also add here that these hooks can completely halt the pipeline, if they return `{stop, NewAcc}`, we will not only stop the current hook as generally `gen_hook` does, but we will also not run the following one, meaning, a stop in `user_send_packet` will also avoid running the subsequent `user_send_*`.